### DISCUSSION TITLE
Next.js integration env files

### DISCUSSION THREAD

**Discussion by runoncedev** [prisma/prisma#7297] [Q&A] [answered]
Hi, I'm using Prisma with Next.js and found some contradicting documentation on each one.

Prisma states that[ `.env` files should not be part of your source control](https://www.prisma.io/docs/concepts/components/prisma-schema#using-env-files):

> Warning: Do not commit your .env files into version control.

But [Next.js states the opossite](https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables):

> Note: .env, .env.development, and .env.production files should be included in your repository as they define defaults. .env*.local should be added to .gitignore, as those files are intended to be ignored. .env.local is where secrets can be stored.

So I ended up with a `.env.local` with all my local environmental variables except for `DATABASE_URL` which has to be on the `.env` file for Prisma to pick it up.

Am I missing something? What is the recommendation on this case?

Thank you.

PS: I think this could be solved by https://github.com/prisma/prisma/issues/3865.

**Comment by ryands17** [comment, marked as answer]
@oncet 👋 

Yes that issue would solve your problem, but as a workaround, you could use something like this in your `package.json` scripts file:

```
"prisma": "dotenv -c $NODE_ENV -- prisma"
```

This will make sure `.env.local` files are picked before `.env` files and based on the `NODE_ENV`, `.env.development` or `.env.production` will be taken.

It's basically your personal preference if you would like to check-in the environment files. Usually production applications environment variables are set in the deployment system rather than using a file based approach but this is fine as well.
